src/models/challenges.d.ts

```diff
@@ -45,6 +45,7 @@ declare namespace socket_api {
             time_per_move: number;
             rengo: boolean;
             rengo_casual_mode: boolean;
+            rengo_auto_start: boolean;
```

Comment:
Is this really boolean?  If that's the case, this line won't render:

https://github.com/online-go/online-go.com/pull/1690/commits/928c708694ba79b5355ab2a1a2012f6e4f610c8f#diff-0213bf671b75d32ee5d114fa9778328a33dff730ace5addd93bc948dff2f0230R1338

However, I suspect this is a number, like the other one :)

Reply:
Eyes like a hawk, this man!

Reply:
... but why didn't things complain?!

Reply:
It's all valid code.  Keep in mind we're just making up types for the API.  No way to actually verify it (unless the backend were written in TS and we used the same .d.ts files)

Reply:
Also, if you wanted a complaint, annotate the variable as string:


<img width="570" alt="Screen Shot 2022-02-03 at 10 11 38 PM" src="https://user-images.githubusercontent.com/25233703/152481181-31e13f35-1477-4538-829d-d7345467cefb.png">

Reply:
^^ I know it's just compiler hints, it's the complaining I was missing.   I think that's an excellent explanation though: if you want types to help you have to type your darn variables darn it.

Reply:
... although then it's a hassle because auto_start is a number 🤣

Reply:
Oh wait yeah hmm.. well I guess explicit `number|string` is better than implicit `boolean|string`...

Reply:
... and if you make it as string then shortcut || doesn't work eh...

Reply:
Ah - I was thinking  `const rengo_auto_start_text: string = C.rengo_auto_start.toString() || "No"`

but you meant `const rengo_auto_start_text: string | number = C.rengo_auto_start || "No".

Right, wake up me.

Reply:
Yep that haha